x/subspaces/keeper/subspaces.go

```diff
@@ -88,9 +88,15 @@ func (k Keeper) DeleteSubspace(ctx sdk.Context, subspaceID uint64) {
 	k.DeleteNextSectionID(ctx, subspaceID)
 	k.DeleteNextGroupID(ctx, subspaceID)
 
-	// Delete all sections
-	k.IterateSubspaceSections(ctx, subspaceID, func(section types.Section) (stop bool) {
-		k.DeleteSection(ctx, section.SubspaceID, section.ID)
+	// Delete all the user grants
+	k.IterateSubspaceUserGrants(ctx, subspaceID, func(grant types.UserGrant) (stop bool) {
+		k.DeleteUserGrant(ctx, grant.SubspaceID, grant.Granter, grant.Grantee)
+		return false
+	})
+
+	// Delete all the group grants
+	k.IterateSubspaceGroupGrants(ctx, subspaceID, func(grant types.GroupGrant) (stop bool) {
+		k.DeleteGroupGrant(ctx, grant.SubspaceID, grant.Granter, grant.GroupID)
 		return false
 	})
```

Comment:
Ideall, this is not necessary. Deleting the sections will cause to delete the groups, which will trigger the deletion of their  grants as well.

Reply:
Follow this idea, `k.IterateSubspaceUserGroups` can be removed as well since it is included in the `k.IterateSubspaceSections`. From my side, using subspace id to remove `UserGroups` and `UserGrants` will have better performance since they are indexed with `SubspaceID` instead of `SectionID` and `GroupID`.